src/request/shard.rs

```diff
@@ -30,21 +36,27 @@ pub trait Shardable {
     type Shard: Send;
 
     fn shards(
-        &self,
+        shard: Self::Shard,
         pd_client: &Arc<impl PdClient>,
     ) -> BoxStream<'static, Result<(Self::Shard, Store)>>;
 
+    fn get_shard(&self) -> Self::Shard;
```

Comment:
I think `get_` had better return a reference. Let the user clone it if necessary.

Reply:
If the shard is computed, `self` doesn't have the ownership, thus not able to return a reference. Anyway, the only usage of get_shard are `MultiRegion` and `PreserveShard` which will always clone it.

Reply:
Then we can name it after something like `get_cloned_shard`

Reply:
I reverted get_shard.